jwst/resample/resample_spec.py

```diff
@@ -750,26 +763,31 @@ def build_interpolated_output_wcs(self, input_models, pixel_scale_ratio=1.0):
 
         native2celestial = RotateNative2Celestial(ra_center_final, dec_center_final, 180)
         undist2sky = tan | native2celestial
-        # find the spatial size of the output - same in x,y
-        if swap_xy:
-            _, x_tan_all = undist2sky.inverse(all_ra, all_dec)
-            pix_to_tan_slope = pix_to_ytan.slope
-        else:
-            x_tan_all, _ = undist2sky.inverse(all_ra, all_dec)
-            pix_to_tan_slope = pix_to_xtan.slope
 
-        x_min = np.amin(x_tan_all)
-        x_max = np.amax(x_tan_all)
-        x_size = int(np.ceil((x_max - x_min) / np.absolute(pix_to_tan_slope)))
+        ## Use all the wcs
+        min_tan_x, max_tan_x, min_tan_y, max_tan_y = self._max_spatial_extent(
+            all_wcs, undist2sky.inverse)
+        diff_y = np.abs(max_tan_y - min_tan_y)
+        diff_x = np.abs(max_tan_x - min_tan_x)
+        pix_to_tan_slope_y = np.abs(pix_to_ytan.slope)
+        slope_sign_y = np.sign(pix_to_ytan.slope)
+        pix_to_tan_slope_x = np.abs(pix_to_xtan.slope)
+        slope_sign_x = np.sign(pix_to_xtan.slope)
+
         if swap_xy:
-            pix_to_ytan.intercept = -0.5 * (x_size - 1) * pix_to_ytan.slope
+            ny = int(np.ceil(diff_y / pix_to_tan_slope_y)) + 1
         else:
-            pix_to_xtan.intercept = -0.5 * (x_size - 1) * pix_to_xtan.slope
+            ny = int(np.ceil(diff_x / pix_to_tan_slope_x)) + 1
+
+        offset_y  = (ny)/2 * pix_to_tan_slope_y
+        offset_x  = (ny)/2 * pix_to_tan_slope_x
+        pix_to_ytan.intercept =  - slope_sign_y * offset_y
+        pix_to_xtan.intercept =  - slope_sign_x * offset_x
 
         # single model: use size of x_tan_array
         # to be consistent with method before
         if len(input_models) == 1:
-            x_size = int(np.ceil(xstop))
+            ny = int(np.ceil(xstop))
```

Comment:
I think we want to remove this IF statement; see general comments.

Reply:
Made this change and it does make the s_regions look better